### PR TITLE
PHPUnit: Change - Install test suite work with wp-cli. (ED-723)

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: .github/workflows/phpunit.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
         uses: actions/checkout@master
       - name: Install Dependencies
         run: |
-          bash bin/install-wp-tests.sh wordpress_test root root localhost ${{ matrix.wordpress_versions }}
+          bash bin/install-wp-cli-tests.sh wordpress_test root root localhost ${{ matrix.wordpress_versions }}
```
Comment (matipojo) [inline .github/workflows/phpunit.yml:20]:
```suggestion
          bash bin/cli-install-wp-tests.sh wordpress_test root root localhost ${{ matrix.wordpress_versions }}
```

File: bin/install-wp-cli-tests.sh [hunk 2]
```diff
@@ -0,0 +1,222 @@
+#!/bin/bash -e
+
+# For running this script in local env, you need to install:
+# zip archive (for mac users "brew install zip")
+# mysql (for mac users https://flaviocopes.com/mysql-how-to-install/)
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:5]: Remove the note about mac users.

File: bin/install-wp-cli-tests.sh [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+DB_NAME=$1
+DB_USER=$2
+DB_PASS=$3
+DB_HOST=${4-localhost}
+WP_VERSION=${5-latest}
+
+# Set default WP_VERSION (if WP_VERSION string is null, that is, has zero length)
+if [ -z $WP_VERSION ] ; then
+    WP_VERSION=latest
+fi
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:21]: It's already set by line 16?

File: bin/install-wp-cli-tests.sh [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+# Set WP params for settings
+#locale=(he_IL en_AU en_CA en_NZ en_ZA en_GB en_US)
+WP_LOCALE="en_GB"
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:25]:
```suggestion
WP_LOCALE="en_US"
```

File: bin/install-wp-cli-tests.sh [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+DB_HOST=${4-localhost}
+WP_VERSION=${5-latest}
+
+# Set default WP_VERSION (if WP_VERSION string is null, that is, has zero length)
+if [ -z $WP_VERSION ] ; then
+    WP_VERSION=latest
+fi
+
+# Set WP params for settings
+#locale=(he_IL en_AU en_CA en_NZ en_ZA en_GB en_US)
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:24]: Remove comment.

File: bin/install-wp-cli-tests.sh [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+WP_USER="test"
+WP_USER_PASS="test"
+WP_USER_EMAIL="user@example.org"
+WP_SITE_NAME="test"
+WP_THEMES="hello-elementor"
+WP_PLUGINS="elementor"
+
+# Save the current working directory in an environment variable.
+INITIAL_WORKING_DIRECTORY=$(pwd)
+echo "$INITIAL_WORKING_DIRECTORY"
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:35]: Remove.

File: bin/install-wp-cli-tests.sh [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+
+# Set paths to directories where you will want to install: wp-cli | wp | wp test suit
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:37]: Don't use the words `you/we` on comments.
```suggestion
# Set paths to directories where to install.
```

File: bin/install-wp-cli-tests.sh [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+WP_TESTS_DIR=${WP_TESTS_DIR-/tmp/wordpress-tests-lib}
+WP_CORE_DIR=${WP_CORE_DIR-/tmp/wordpress/}
+
+# Set paths to current plugins and them tests directories
+CURRENT_PLUGIN="${WP_CORE_DIR}wp-content/plugins/elementor"
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:42]:
```suggestion
CURRENT_PLUGIN="elementor"
CURRENT_PLUGIN_DIR="${WP_CORE_DIR}wp-content/plugins/${CURRENT_PLUGIN}"
```

File: bin/install-wp-cli-tests.sh [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+ 		mkdir -p "$WP_CORE_DIR"
+	fi
+
+	cd "$WP_CORE_DIR"
+	# Download and install wp-cli
+	curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
+	# Set the permissions to make it executable
+	chmod +x wp-cli.phar
+	sudo mv wp-cli.phar /usr/local/bin/wp
+  wp --info
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:64]: Remove.

File: bin/install-wp-cli-tests.sh [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+}
+
+download_wp_core(){
+	# Download the WordPress core files [--skip-content - Download WP without the default themes and plugins.]
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:68]:
```suggestion
	# Download the WordPress core files [--skip-plugins - Download WP without the default plugins.]
```

File: bin/install-wp-cli-tests.sh [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+config(){
+	# Create the wp-config file
+	wp config create --dbname="$DB_NAME" --dbuser="$DB_USER" --dbpass="$DB_PASS"
+}
+
+install(){
+	# Create database
+	wp db create
+
+	# parse DB_HOST for port or socket references
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:81]:
```suggestion
	# Parse DB_HOST for port or socket references
```

File: bin/install-wp-cli-tests.sh [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+			EXTRA="--socket=$DB_SOCK_OR_PORT"
+		elif ! [ -z $DB_HOSTNAME ] ; then
+			EXTRA="--host=$DB_HOSTNAME --protocol=tcp"
+		fi
+	fi
+
+	# Optimizes the database
+	wp db optimize --dbuser="$DB_USER" --dbpass="$DB_PASS" "$EXTRA"
+
+	# install WordPress
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:100]:
```suggestion
	# Install WordPress
```

File: bin/install-wp-cli-tests.sh [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+	wp core install --url="http://testelementor" --title="$WP_SITE_NAME" --admin_user="$WP_USER" --admin_password="$WP_USER_PASS" --admin_email="$WP_USER_EMAIL"
+
+	download https://raw.github.com/markoheijnen/wp-mysqli/master/db.php $WP_CORE_DIR/wp-content/db.php
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:103]: Explain

File: bin/install-wp-cli-tests.sh [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+		sed $ioption "s/yourusernamehere/$DB_USER/" "$WP_TESTS_DIR"/wp-tests-config.php
+		sed $ioption "s/yourpasswordhere/$DB_PASS/" "$WP_TESTS_DIR"/wp-tests-config.php
+		sed $ioption "s|localhost|${DB_HOST}|" "$WP_TESTS_DIR"/wp-tests-config.php
+	fi
+}
+
+create_plugins_folder(){
+	if [ ! -d "${WP_CORE_DIR}wp-content/plugins" ]; then
+		cd "${WP_CORE_DIR}wp-content/"
+		mkdir plugins
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:151]:
```suggestion
		mkdir "${WP_CORE_DIR}wp-content/plugins"
```

File: bin/install-wp-cli-tests.sh [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+	fi
+}
+
+install_plugins(){
+	create_plugins_folder
+	# Create symlink to current path
+	ln -s "${INITIAL_WORKING_DIRECTORY}" "${CURRENT_PLUGIN}"
+	# After running install_test_suite need to return to directory $WP_CORE_DIR where wp-cli installed
+	cd "${WP_CORE_DIR}"
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:160]: Move to end of  `install_test_suite`

File: bin/install-wp-cli-tests.sh [hunk 16]
```diff
@@ -0,0 +1,222 @@
...
+
+	# Delete the $WP_TESTS_DIR directory and all its contents, including any subdirectories and files
+	if [ -d "$WP_TESTS_DIR" ]; then
+    rm -rf "$WP_TESTS_DIR"
+    if [ ! -d "$WP_TESTS_DIR" ]; then
+      echo "Directory $WP_TESTS_DIR is deleted."
+    else
+		  echo "Directory $WP_TESTS_DIR does not delete."
+		fi
+	fi
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:194]: Fix indentations

File: bin/install-wp-cli-tests.sh [hunk 17]
```diff
@@ -0,0 +1,222 @@
...
+}
+
+delete_symlink(){
+	# Check if file exists and is a symbolic link (same as -h command)
+	if [ -L "${CURRENT_PLUGIN}" ]; then
+		unlink "${CURRENT_PLUGIN}"
+		echo "deleted symlink ${CURRENT_PLUGIN}"
```
Comment (matipojo) [inline bin/install-wp-cli-tests.sh:201]: All debug/log data should be printed via function that print with a format like 
`Date [info|warning|error] - Message.`

File: tests/imgcomp/elementor-17-2020-11-16.json [hunk 18]
```diff
@@ -0,0 +1 @@
...ile":{"unit":"em","size":""},"content_typography_letter_spacing_tablet":{"unit":"px","size":""},"content_typography_letter_spacing_mobile":{"unit":"px","size":""},"name_typography_font_size_tablet":{"unit":"px","size":""},"name_typography_font_size_mobile":{"unit":"px","size":""},"name_typography_line_height_tablet":{"unit":"em","size":""},"name_typography_line_height_mobile":{"unit":"em","size":""},"name_typography_letter_spacing_tablet":{"unit":"px","size":""},"name_typography_letter_spacing_mobile":{"unit":"px","size":""},"job_typography_font_size_tablet":{"unit":"px","size":""},"job_typography_font_size_mobile":{"unit":"px","size":""},"job_typography_line_height_tablet":{"unit":"em","size":""},"job_typography_line_height_mobile":{"unit":"em","size":""},"job_typography_letter_spacing_tablet":{"unit":"px","size":""},"job_typography_letter_spacing_mobile":{"unit":"px","size":""},"_padding_mobile":{"unit":"px","top":"0","right":"0","bottom":"30","left":"0","isLinked":""}},"elements":[],"isInner":false,"widgetType":"testimonial","elType":"widget"}],"isInner":"1","elType":"column"}],"isInner":"1","elType":"section"}],"isInner":"","elType":"column"}],"isInner":"","elType":"section"}]}
```
Comment (matipojo) [inline tests/imgcomp/elementor-17-2020-11-16.json:1]: Remove file.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: bin/install-wp-cli-tests.sh (hunk 5 above)
Comment (efelementor) [reply bin/install-wp-cli-tests.sh]: fixed

File: bin/install-wp-cli-tests.sh (hunk 4 above)
Comment (efelementor) [reply bin/install-wp-cli-tests.sh]: fixed

File: bin/install-wp-cli-tests.sh (hunk 3 above)
Comment (efelementor) [reply bin/install-wp-cli-tests.sh]: Fixed

File: bin/install-wp-cli-tests.sh (hunk 2 above)
Comment (efelementor) [reply bin/install-wp-cli-tests.sh:5]: Fixed

File: .github/workflows/phpunit.yml (hunk 1 above)
Comment (efelementor) [reply .github/workflows/phpunit.yml:20]: Not fixed because we also install the cli

File: bin/install-wp-cli-tests.sh (hunk 14 above)
Comment (efelementor) [reply bin/install-wp-cli-tests.sh]: Fixed

File: bin/install-wp-cli-tests.sh (hunk 15 above)
Comment (efelementor) [reply bin/install-wp-cli-tests.sh]: Fixed

File: bin/install-wp-cli-tests.sh (hunk 16 above)
Comment (efelementor) [reply bin/install-wp-cli-tests.sh]: Fixed

File: bin/install-wp-cli-tests.sh (hunk 17 above)
Comment (efelementor) [reply bin/install-wp-cli-tests.sh]: Fixed I added print_msg function as you wish

File: tests/imgcomp/elementor-17-2020-11-16.json (hunk 18 above)
Comment (efelementor) [reply tests/imgcomp/elementor-17-2020-11-16.json]: Fixed

File: .github/workflows/phpunit.yml (hunk 1 above)
Comment (efelementor) [reply .github/workflows/phpunit.yml:20]: Fixed

File: bin/install-wp-cli-tests.sh (hunk 5 above)
Comment (efelementor) [reply bin/install-wp-cli-tests.sh]: Fixed

File: bin/install-wp-cli-tests.sh (hunk 4 above)
Comment (efelementor) [reply bin/install-wp-cli-tests.sh]: Fixed

**Review by matipojo — CHANGES_REQUESTED** [review]

File: bin/cli-install-wp-tests.sh [hunk 19]
```diff
@@ -0,0 +1,223 @@
...
+
+# Save the current working directory in an environment variable.
+INITIAL_WORKING_DIRECTORY=$(pwd)
+echo "$INITIAL_WORKING_DIRECTORY"
+
+# Set paths to directories where to install.
+WP_TESTS_DIR=${WP_TESTS_DIR-/tmp/wordpress-tests-lib}
+WP_CORE_DIR=${WP_CORE_DIR-/tmp/wordpress/}
+
+# Set paths to current plugins and them tests directories
```
[the remark this answers is not in the document]
Comment (efelementor) [reply bin/cli-install-wp-tests.sh:35]: Fixed

File: bin/cli-install-wp-tests.sh [hunk 20]
```diff
@@ -0,0 +1,223 @@
...
+		fi
+	fi
+
+	# Optimizes the database
+	wp db optimize --dbuser="$DB_USER" --dbpass="$DB_PASS" "$EXTRA"
+
+	# Install WordPress
+	wp core install --url="http://testelementor" --title="$WP_SITE_NAME" --admin_user="$WP_USER" --admin_password="$WP_USER_PASS" --admin_email="$WP_USER_EMAIL"
+
+	# If you want to switch to mysqli you put define( 'WP_USE_EXT_MYSQL', false ); in wp-config and WP core will use mysqli without any plugin.
```
[the remark this answers is not in the document]
Comment (efelementor) [reply bin/cli-install-wp-tests.sh:103]: Fixed

File: elementor [hunk 21]
```diff
@@ -0,0 +1 @@
+/Users/etayf/Local Sites/localelementorcom/app/public/wp-content/plugins/elementor
```
[the remark this answers is not in the document]
Comment (efelementor) [reply elementor]: Fixed

File: bin/cli-install-wp-tests.sh [hunk 22]
```diff
@@ -0,0 +1,223 @@
+#!/bin/bash -e
+
+# For running this script in local env, you need to install:
```
Comment (matipojo) [inline bin/cli-install-wp-tests.sh:3]:
```suggestion
# For running this script in local env, it requires to install:
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: bin/install-wp-cli-tests.sh (hunk 17 above)
Comment (matipojo) [reply bin/install-wp-cli-tests.sh]: You should identify each message type (`info|warning|error`)

**Review by matipojo — CHANGES_REQUESTED** [review]
Please fix all issues + fix merge conflict

File: bin/cli-install-wp-tests.sh (hunk 22 above)
Comment (etayf) [reply bin/cli-install-wp-tests.sh]: Fixed

Comment (etayf) [reply bin/cli-install-wp-tests.sh]: Fixed

File: bin/test-functions.sh [hunk 23]
```diff
@@ -0,0 +1,201 @@
...
+}
+
+install_themes() {
+	# Install the theme
+	wp theme install "$wp_themes" --activate
+}
+
+
+#############################################################################################
+####################### The Below Functions Clean The Local Tests Env #######################
```
Comment (matipojo) [inline bin/test-functions.sh:148]: Remove all unnecessary  `#` +lower case when needed

File: bin/test-functions.sh [hunk 24]
```diff
@@ -0,0 +1,201 @@
+#!/bin/bash -e
+
+#############################################################################################
```
Comment (matipojo) [inline bin/test-functions.sh:3]: Remove all unnecessary  `#`

File: bin/test-functions.sh [hunk 25]
```diff
@@ -0,0 +1,201 @@
...
+}
+
+wp_cli_cache_clear() {
+	# Clears the internal cache
+	wp cli cache clear
+}
+
+#############################################################################################
+################ The Below Function Run Functions Clean The Local Tests Env #################
+#############################################################################################
```
Comment (matipojo) [inline bin/test-functions.sh:193]: Remove all unnecessary # +lower case when needed

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
Fix conflict.

**Pull request closed without merging**